k8s-plugin-keel/src/main/kotlin/com/amazon/spinnaker/keel/k8s/verificationEvaluator/K8sJobEvaluator.kt

```diff
@@ -0,0 +1,182 @@
+        val response = runBlocking(Dispatchers.IO) {
+            try {
+                orcaService.getOrchestrationExecution(taskId, context.deliveryConfig.serviceAccount)
+            } catch (e: HttpException) {
+                if (e.code() == 404) {
+                    log.warn("task with task id $taskId not found. metadata: ${oldState.metadata} verification: $verification")
+                    null
+                } else {
+                    throw e
+                }
+            }
+        }
+        log.debug("response from orca for task $taskId: $response")
+        response?.let {
+            return when {
+                response.status.isSuccess() -> oldState.copy(status = ConstraintStatus.PASS).also {
+                    log.info("verification was successful. taskId: $taskId")
+                }
+                response.status.isIncomplete() -> oldState.copy(ConstraintStatus.PENDING).also {
+                    log.debug("verification still in progress. taskId: $taskId")
+                }
+                else -> oldState.copy(ConstraintStatus.FAIL).also {
+                    log.warn("verification failed. taskId: $taskId")
+                }
+            }
+        } ?: return oldState.copy(status = ConstraintStatus.FAIL)
+    }
+
+    override fun start(context: ArtifactInEnvironmentContext, verification: Verification): Map<String, Any?> {
+        require(verification is K8sJobVerification) {
+            "verification class must be ${K8sJobVerification::class.simpleName}. received: ${verification.javaClass.simpleName}"
+        }
+        log.info("launching verification K8s job for ${context.deliveryConfig.application} env: ${context.environmentName}")
+        val (job, metadata) = generateOrcaK8sJob(
+            context.deliveryConfig.application,
+            context.environmentName,
+            verification
+        )
+
+        log.info("this job will be launched in namespace, ${metadata[NAMESPACE]}. Name: ${metadata["jobName"]}")
+        return runBlocking {
+            val launcherResponse = taskLauncher.submitJob(
+                user = context.deliveryConfig.serviceAccount,
+                application = context.deliveryConfig.application,
+                notifications = emptySet(),
+                subject = "verify environment, ${context.environmentName}, for ${context.deliveryConfig.application}",
+                description = "verifying ${context.version} in environment ${context.environmentName} with K8s $VERIFICATION_K8S_JOB_V1",
+                correlationId = verification.id,
+                stages = listOf(job),
+                type = VERIFICATION,
+                artifacts = emptyList(),
+                parameters = emptyMap()
+            )
+            metadata += ("taskId" to launcherResponse.id)
+            metadata + ("taskName" to launcherResponse.name)
```

Comment:
let's add comments on how this `metadata` is used. not very clear just by looking at the code.